esp/xslt/esp_service_xml2xsd.xslt

```diff
@@ -153,6 +153,8 @@
                     </xsl:when>
                     <xsl:when test="@type='int64'">xsd:<xsl:value-of select="'long'"/>
                     </xsl:when>
+                    <xsl:when test="@type='TpQueryType'">xsd:<xsl:value-of select="'string'"/>
```

Comment:
@kenrowland This is actually covering up a bug in ws_topology.ecm.  TpQueryType was meant to be defined as an enumerated string.  But it was incorrectly defined and incorrectly used.

You could change the definition of TpQueryType from ESPstruct to ESPenum and then make use of it, BUT that may cause backward compatibility issues.  
The alternative for maximum backward compatibility would be just to change it to a string.

@wangkx when you were changing strings to enums how did we handle backward compatability?

Anyway I wouldn't make this change, one way or the other I would fix the bad item in the ecm.

Reply:
In ecm file, add a new variable and 'depr_ver' the old one. If the variables are in a request, make sure that the new variable is read first. If not found, read the old one for backward compatibility.

Reply:
@kenrowland for now I would just declare it as a string but add a comment about the two expected values (from the struct definition).  It's no worse or less informative than what is there now.